src/components/interactive-builder/modals/question/question-form/rendering-types/inputs/select/select-answers.component.tsx

```diff
@@ -124,6 +126,46 @@ const SelectAnswers: React.FC = () => {
     return [...conceptAnswerItems, ...additionalAnswers];
   }, [concept?.answers, formField.questionOptions?.answers]);
 
+  useEffect(() => {
+    if (!concept?.answers?.length) return;
+    const validateAnswers = async () => {
+      if (!answerItems.length) {
+        setInvalidAnswerFound(false);
+        return;
+      }
+
+      const originalAnswerIds = new Set(concept?.answers?.map((ans) => ans.uuid) || []);
+
+      let allValid = true;
+      const lookupCache = new Map<string, boolean>();
+      const uniqueAnswers = Array.from(new Map(answerItems.map((a) => [a.id, a])).values());
+
+      for (const answer of uniqueAnswers) {
+        if (originalAnswerIds.has(answer.id)) {
+          continue;
+        } else {
+          const url = `${restBaseUrl}/concept/${answer.id}?v=full`;
+          try {
+            const res = await openmrsFetch(url);
+            const isValid = res?.data?.uuid === answer.id;
+
+            lookupCache.set(answer.id, isValid);
+            if (!isValid) {
+              allValid = false;
+              break;
+            }
+          } catch (error) {
+            allValid = false;
+            break;
+          }
+        }
+      }
+      setInvalidAnswerFound(!allValid);
+    };
+
+    validateAnswers();
```

Comment:
Since you are calling this function inside useEffect, the function should be defined outside the `useEffect` hook.